fuel-core/src/schema/dap.rs

```diff
@@ -192,4 +192,272 @@ impl DapMutation {
 
         Ok(result)
     }
+
+    #[cfg(not(feature = "debug"))]
+    async fn set_single_stepping(
+        &self,
+        _ctx: &Context<'_>,
+        _id: ID,
+        _enable: bool,
+    ) -> async_graphql::Result<bool> {
+        Err(async_graphql::Error::new(
+            "Feature 'debug' is not compiled in",
+        ))
+    }
+
+    #[cfg(feature = "debug")]
+    async fn set_single_stepping(
+        &self,
+        ctx: &Context<'_>,
+        id: ID,
+        enable: bool,
+    ) -> async_graphql::Result<bool> {
+        trace!("Set single stepping to {} for VM {:?}", enable, id);
+
+        let mut locked = ctx.data_unchecked::<GraphStorage>().lock().await;
+        let vm = locked
+            .vm
+            .get_mut(&id)
+            .ok_or_else(|| async_graphql::Error::new("VM not found"))?;
+
+        vm.set_single_stepping(enable);
+        Ok(enable)
+    }
+
+    #[cfg(not(feature = "debug"))]
+    async fn set_breakpoint(
+        &self,
+        _ctx: &Context<'_>,
+        _id: ID,
+        _breakpoint: self::gql_types::Breakpoint,
+    ) -> async_graphql::Result<bool> {
+        Err(async_graphql::Error::new(
+            "Feature 'debug' is not compiled in",
+        ))
+    }
+
+    #[cfg(feature = "debug")]
+    async fn set_breakpoint(
+        &self,
+        ctx: &Context<'_>,
+        id: ID,
+        breakpoint: self::gql_types::Breakpoint,
+    ) -> async_graphql::Result<bool> {
+        trace!("Continue exectuon of VM {:?}", id);
+
+        let mut locked = ctx.data_unchecked::<GraphStorage>().lock().await;
+        let vm = locked
+            .vm
+            .get_mut(&id)
+            .ok_or_else(|| async_graphql::Error::new("VM not found"))?;
+
+        vm.set_breakpoint(breakpoint.into());
+        Ok(true)
+    }
+
+    async fn start_tx(
+        &self,
+        ctx: &Context<'_>,
+        id: ID,
+        tx_json: String,
+    ) -> async_graphql::Result<self::gql_types::RunResult> {
+        trace!("Spawning a new VM instance");
+
+        dbg!(&tx_json);
```

Comment:
why not using `tracing` here?

Reply:
Just a left-over debug print, slipped past me. Removed.

We probably should deny `dbg_macro` lint in the CI.